examples/07_grid_searching_with_the_tablevectorizer.py

```diff
@@ -0,0 +1,144 @@
+    high_card_cat_transformer=MinHashEncoder(),
+)
+tv.fit(X)
+
+pprint(tv.transformers_)
+
+###############################################################################
+# If we want to modify what we classify as a high cardinality categorical
+# column, we can tweak the ``cardinality_threshold`` parameter.
+# Check out the |TableVectorizer|'s doc for more information.
+#
+# Also have a look at the other types of columns supported by default!
+#
+# Using a custom Transformer for a specific column
+# ................................................
+#
+# Say we wanted to use a |MinHashEncoder| instead of the default |GapEncoder|,
+# but only for the column ``department_name``.
+# The best way to apply a column-specific transformer is to use the
+# dedicated parameter ``specific_transformers``:
+
+tv = TableVectorizer(specific_transformers=[(MinHashEncoder(), ["department_name"])])
+tv.fit(X)
+
+pprint(tv.transformers_)
+
+###############################################################################
+# Here, for simplicity, we used the unnamed 2-tuple syntax.
+#
+# You can also give a name to the assignment, as we will see in the next
+# section.
+#
+# Grid-searching with the TableVectorizer
+# ---------------------------------------
+#
+# Grid-searching the encoders' hyperparameters contained in the
+# |TableVectorizer| is easy!
+# For that, we use the dunder separator, which indicates a nesting layer.
+# That means that for tuning the parameter ``n_components`` of the
+# `GapEncoder` saved in the |TableVectorizer| attribute
+# ``high_card_cat_transformer``, we use the syntax
+# ``tablevectorizer__high_card_cat_transformer__n_components``.
+#
+# For the column-specific transformers we specify, it is recommended to
+# use the 3-tuple syntax, which enables giving a name to the assignment
+# (here ``mh_dep_name``).
+
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import make_pipeline
+from sklearn.ensemble import HistGradientBoostingClassifier
+from skrub import GapEncoder
+
+pipeline = make_pipeline(
+    TableVectorizer(
+        high_card_cat_transformer=GapEncoder(),
+        specific_transformers=[
+            ("mh_dep_name", MinHashEncoder(), ["department_name"]),
+        ],
+    ),
+    HistGradientBoostingClassifier(),
+)
+
+params = {
+    "tablevectorizer__high_card_cat_transformer__n_components": [10, 30, 50],
+    "tablevectorizer__mh_dep_name__n_components": [25, 50],
+}
+
+grid_search = GridSearchCV(pipeline, param_grid=params)
+
+###############################################################################
+# Conclusion
+# ----------
+#
+# In this notebook, we saw how to better customize the |TableVectorizer| so
+# it fits all your needs!
+#
+# If you've got any improvement ideas, please open a feature request on
+# `GitHub <https://github.com/skrub-data/skrub/issues/new?labels=enhancement&template=feature_request.yml>`_!
+#
+# We are always happy to see new suggestions from the community :)
```

Comment:
Nice! Maybe we should add this at the end of every example :)

Reply:
Good idea, maybe this can be added to the template later on, as a callout box!

Reply:
Opened #689 in this regard